src/scicookie/{{cookiecutter.project_slug}}/build-system/setuptools-pyproject.toml

```diff
@@ -110,6 +110,9 @@ dependencies = [
 {% endif %}
 ]
 
+[tool.setuptools.dynamic]
+version = {attr = "{{ cookiecutter.project_slug }}.__version__"}
```

Comment:
I saw that the `__version`, `version`, and `get_version()` reference each other and as a result return the same string (i.e., `'0.1.0'` in this case). I went ahead with the [PEP 440](https://packaging.python.org/en/latest/guides/single-sourcing-package-version/) recommended specifier.